src/include/daos_types.h

```diff
@@ -226,6 +233,29 @@ typedef struct {
 	uint64_t	hi;
 } daos_obj_id_t;
 
+#define DAOS_UUID_STR_SIZE 37	/* 36 + 1 for '\0' */
+
+static inline bool
+daos_is_valid_uuid_string(const char *uuid)
+{
+	const char	*p;
+	int		 len = DAOS_UUID_STR_SIZE - 1; /* Not include the terminated '\0' */
+	int		 i;
+
+	if (strnlen(uuid, len) != len)
```

Comment:
This is moved from daos_label_is_valid(). Here is the original comment (via the commit 91da60190d17c486b65d0c15ffad3069e65f0a5b) although I do not know how much the overhead will be.

```
/** Implement the check directly to avoid uuid_parse() overhead */
```